lib/src/retriever.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import "dart:collection";
+
+import "package:pool/pool.dart";
+import "package:async/async.dart";
+
+/// Handles rate-limited scheduling of tasks.
```

Comment:
The name feels unrelated to the description. Should we rename from `Retriever` to `RateLimitedScheduler`? Are we mixing 2 concepts (retrieving and rate limiting) that should be split?

Reply:
I updated the name - it is better.

The retrieving behavior is already abstracted, not sure how it would be split further.